aio/content/examples/comparing-observables/src/promises.ts

```diff
@@ -1,25 +1,43 @@
-// #docregion promise
-// initiate execution
-const promise = new Promise<number>((resolve, reject) => {
-  // Executer fn...
-});
-
-promise.then(value => {
-  // handle result here
-});
-
-// #enddocregion promise
-
-// #docregion chain
-
-promise.then(v => 2 * v);
-
-// #enddocregion chain
-
-// #docregion error
-
-promise.then(() => {
-  throw Error('my error');
-});
-
-// #enddocregion error
+// #docplaster
+
+export function docRegionPromise(console: Console) {
+  // #docregion promise
+  // initiate execution
+  const promise = new Promise<number>((resolve, reject) => {
+    // Executer fn...
+    // #enddocregion promise
+    // The below is used in the unit tests.
+    resolve(2);
+    // #docregion promise
+  });
+
+  promise.then(value => {
+    // handle result here
+    // #enddocregion promise
+    // The below is used in the unit tests.
+    console.log(value);
+    // #docregion promise
+  });
+
+  // #enddocregion promise
+
+  // #docregion chain
+
+  promise.then(v => 2 * v);
+
+  // #enddocregion chain
```

Comment:
I just realized that in order for this to be testable, we need to refactor the code a bit (and also use the "trick" from `docRegionError()` below).

```suggestion
  let promise = new Promise<number>((resolve, reject) => {
    // Executer fn...
    // #enddocregion promise
    // The below is used in the unit tests.
    resolve(2);
    // #docregion promise
  });

  // #enddocregion promise
  promise =
  // #docregion chain

  promise.then(v => 2 * v);

  // #enddocregion chain
  promise =
  // docregion promise
  promise.then(value => {
    // handle result here
    // #enddocregion promise
    // The below is used in the unit tests.
    console.log(value);
    // #docregion promise
  });

  // #enddocregion promise
  return promise;
```

[1]: https://github.com/angular/angular/blob/82d200a26f363f564a1f743bac92aa132638c7b3/aio/content/examples/comparing-observables/src/promises.ts#L33-L39

Reply:
I guess at line 28, we need to return `value` as well, otherwise, the test will not pass, right?

Reply:
I have reordered the `then()` calls, so you can now assert that `console.log()` is called with `4` (i.e. the transformed value). You can also update the `docRegion...` function to accept the value to resolve the initial promise with.